### PR TITLE
Status endpoint

### PR DESCRIPTION
**Pull request by runnersaw**
Untested, just a framework for keeping track of the status of analysis of different videos.

### CODE REVIEW

**Comment by jabb1123** [comment]
Was this resolved?

File: app/app.py
Comment (youralien) [inline app/app.py]: Shouldn't the handler be `TrajectoryAnalysisStatusHandler`

File: app/app.py
Comment (youralien) [inline app/app.py]: Same here. `SafetyAnalysisStatusHandler`

File: app/utils/statusHelper.py
Comment (youralien) [inline app/utils/statusHelper.py]: I'm not liking the indents on this file.  Are you sure this is 4 spaces per tab?

File: app/utils/statusHelper.py
Comment (youralien) [inline app/utils/statusHelper.py]: When does `Status` class get called?  Currently its unused.

File: app/utils/statusHelper.py
Comment (youralien) [inline app/utils/statusHelper.py]: Where does `StatusHelper` get used?   If this PR is a WIP, then I expect these classes to be used in the StatusHandler post/get methods.  Am I right about this guess?

**Review by youralien — DISMISSED** [review]

**Comment by runnersaw** [comment]
There's a few comments relating to the purpose of this pull request, so I'll just reply here to explain a little bit. 

The idea of this pull request was to set up a framework for keeping track of the status of different analyses. As such, this PR creates handlers for routes related to checking the status of analyses (but doesn't actually handle it yet). The StatusHandler class and related enum is intended to keep track of the status of these analyses. 

My intention was to create a shared status handler (a singleton) that keeps track of the status of analyses using a dictionary. The intention was for it to be easy to update the status of different videos simply by importing it and calling `sharedStatusHandler.set_status(my_uuid, Status.TRAJECTORY_ANALYSIS)`.

This snippet would have to be scattered into other code in order to ensure that the status of a given video is always correct.

Then, the job of the status handlers would be to get the UUID from the request, look up its status using `sharedStatusHandler.get_status(my_uuid)`, and return that to the user.  I didn't implement this code ATM, although this could be done now.

**Comment by jabb1123** [comment]
I reviewed the changes. Seems like you deal with all of Ryan's concerns. 🚢  🇮🇹

File: app/handlers/trajectoryAnalysisStatus.py
Comment (youralien) [inline app/handlers/trajectoryAnalysisStatus.py]: How is TrajectoryAnalysisStatusHandler and SafetyAnalysisStatusHandler different?  As I see it, the user sends a 'videoId' and the sharedStatusHelper tells them where it is in the process. I can see one handler doing all of that, and not needing to specify two different endpoints.

File: app/utils/statusHelper.py
Comment (youralien) [inline app/utils/statusHelper.py]: How would we use this singleton to set status, say, in the upload.py file?  Would importing this singleton into that file share the same state as an import of the singleton in another file?

Comment (runnersaw) [reply app/utils/statusHelper.py]: Python loads modules by name only once, so this sharedStatusHelper is only created once and is the same when imported anywhere. See http://stackoverflow.com/a/13331066/5224196

File: app/handlers/trajectoryAnalysisStatus.py
Comment (runnersaw) [reply app/handlers/trajectoryAnalysisStatus.py]: Yes, these can be the same

**Comment by runnersaw** [comment]
Now there's no singleton. The StatusHelper class stores and retrieves project status from the configuration file.

**Comment by jabb1123** [comment]
Good stuff here. Wait for another review but I give it a 🚢 🇮🇹

File: app/traffic_cloud_utils/emailHelper.py
Comment (dcelik) [inline app/traffic_cloud_utils/emailHelper.py]: Are these meant to be here? Won't this break the code

**Review by jabb1123 — APPROVED** [review]
Good to see the fixes to Email helper

**Review by dcelik — APPROVED** [review]

**Pull request merged**
